modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java

```diff
@@ -62,32 +58,50 @@ public String transform(String html) throws PortalException {
 			return null;
 		}
 
-		if (!html.contains("/documents/") || !html.contains("<img")) {
+		if (!html.contains("/documents/") ||
+			!html.contains(_OPEN_TAG_TOKEN_IMG)) {
+
 			return html;
 		}
 
-		Document document = _parseDocument(html);
+		StringBundler sb = new StringBundler();
 
-		for (Element imgElement : document.select("img:not(picture > img)")) {
-			String imgElementString = imgElement.toString();
+		int lastPos = 0;
 
-			String replacement = _transform(
-				imgElementString, imgElement.attr("src"));
+		while (lastPos < html.length()) {
+			int pictureStart = html.indexOf(_OPEN_TAG_TOKEN_PICTURE, lastPos);
 
-			imgElement.replaceWith(_parseNode(replacement));
```

Comment:
It still failing as the blocker said:

<img width="1571" alt="image" src="https://github.com/liferay-lima/liferay-portal/assets/47524751/895c6c89-5717-4990-a2d6-f26e687e51d9">


The problem, as I see is that this line is not replaced with on the new code, so the 

```
_transform(
					html.substring(imgStart, imgEnd),
					html.substring(quotePos + 1, srcEnd))
```

is generating 
`<img alt="" class="navbar-logo" data-lfr-editable-id="imageLogo" data-lfr-editable-type="image" src="/documents/d/site/masterclass_logo-svg-1?download=true" width="162">?download=true`

instead of 

`<img alt="" class="navbar-logo" data-lfr-editable-id="imageLogo" data-lfr-editable-type="image" src="/documents/d/site/masterclass_logo-svg-1?download=true" width="162">`